### PR TITLE
RUBY-1868 actually close local clients after each test

### PR DESCRIPTION
**Pull request by p-mongo**
https://jira.mongodb.com/browse/RUBY-1868

### CODE REVIEW

**Comment by p-mongo** [comment]
It appears that the test suite was not actually closing local clients after every example, which caused background threads to stick around for longer than they should have.

File: spec/mongo/operation/update/op_msg_spec.rb [hunk 1]
```diff
@@ -160,14 +160,19 @@
...
-            authorized_client.cluster.servers.first.monitor.stop!
-            allow(authorized_primary.features).to receive(:sessions_enabled?).and_return(false)
+            authorized_primary.monitor.stop!
           end
 
           it 'creates the correct OP_MSG message' do
-            authorized_client.command(ping:1)
-            expect(Mongo::Protocol::Msg).to receive(:new).with([], {}, expected_global_args, expected_payload_1)
-            op.send(:message, authorized_primary)
+            RSpec::Mocks.with_temporary_scope do
```
Comment (egiurleo) [inline spec/mongo/operation/update/op_msg_spec.rb]: Could you explain why you use the `Rspec::Mocks.with_temporary_scope` here? My assumption would be that any doubles or mocks set up in this example would be cleared at the end of the example, but it's possible there's something else going on that I don't understand.

Comment (p-mongo) [reply spec/mongo/operation/update/op_msg_spec.rb]: The purpose of the scope is to limit assertions to the operations being tested only, not to create the first scope.

`after` callbacks close local clients (this was meant to be the case prior to this PR also, but I omitted the call to do so erroneously). When clients are closed they can perform network operations, for example `endSessions` calls to clear session pools. RSpec's assertion scope by default covers all `before`/`after` callbacks, therefore assertions defined in this example fail when an `after` callback runs `endSessions` and instantiates additional `Protocol::Msg` objects.

Comment (egiurleo) [reply spec/mongo/operation/update/op_msg_spec.rb]: Okay, I see. So the purpose of the `Rspec::Mocks.with_temporary_scope` is to create a scope that does not include `before/after`, correct?

Comment (p-mongo) [reply spec/mongo/operation/update/op_msg_spec.rb]: That's right.

**Pull request merged**